src/test/java/com/google/firebase/database/connection/WebsocketConnectionTest.java

```diff
@@ -112,7 +112,7 @@ public void testReceiveMultipleFrames() {
   public void testIncomingMessageIOError() {
     MockClientFactory clientFactory = new MockClientFactory();
     ImmutableMap<String, Object> data = ImmutableMap.<String, Object>of("key", "value");
-    Mockito.doThrow(IOException.class).when(clientFactory.delegate).onMessage(data);
+    Mockito.doThrow(ClassCastException.class).when(clientFactory.delegate).onMessage(data);
```

Comment:
Lets be generic and make it a `RuntimeException`

Reply:
this won't be ok ... the reason why I added ClassCastException was the way onMessage is implemented.

```
    // Decode JSON
    String combined = buffer.combine();
    try {
      Map<String, Object> decoded = JsonMapper.parseJson(combined);
      logger.debug("{} Parsed complete frame: {}", label, decoded);
      delegate.onMessage(decoded);
    } catch (IOException e) {
      logger.error("{} Error parsing frame: {}", label, combined, e);
      closeAndNotify();
    } catch (ClassCastException e) {
      logger.error("{} Error parsing frame (cast error): {}", label, combined, e);
      closeAndNotify();
    }
```
Point of unit test is to get into closeAndNotify() method.

Reply:
Hrm. I expected this diff to go away. Am I missing something?

Reply:
Mockito can no longer throw IOException for method that does not have throws. It goes again Java language. Mockito used to allow this, but latest version does not (rightfully).

ERROR:
org.mockito.exceptions.base.MockitoException:  Checked exception is invalid for this method! Invalid: java.io.IOException

  interface Delegate {
    void onMessage(Map<String, Object> message);
    void onDisconnect(boolean wasEverConnected);
  }

onMessage does not throw IOException:
Mockito.doThrow(IOException.class).when(clientFactory.delegate).onMessage(data);

I can revert, but then unit test is gonna fail. This fix is related to Mockito version upgrade (for java 11) and does NOT have to do with JsonMapper changes.

a) leave as it is
b) add throws IOException to onMessage for the purpose of unit test (and catch it in all the places it affects)
c) remove the test because it mocks the wrong part
d) adding powermock dep and mock static JsonMapper method

Reply:
I have take the option e) ... 
this code was broken anyway so I have changed it to not even have a mock.